applications/nfc/scenes/nfc_scene_saved_menu.c

```diff
@@ -62,7 +74,11 @@ bool nfc_scene_saved_menu_on_event(void* context, SceneManagerEvent event) {
             scene_manager_next_scene(nfc->scene_manager, NfcSceneDelete);
             consumed = true;
         } else if(event.event == SubmenuIndexInfo) {
-            scene_manager_next_scene(nfc->scene_manager, NfcSceneDeviceInfo);
+            if(nfc->dev->format == NfcDeviceSaveFormatMifareDesfire) {
+                scene_manager_next_scene(nfc->scene_manager, NfcSceneReadMifareDesfireSuccess);
```

Comment:
I think it's better to show DESFire cards Info in NfcSceneDeviceInfo, or make another NfcSceneMifareDesfireInfo scene. If I go to  NfcSceneReadMifareDesfireSuccess scene from **Saved cards**, I don't expect to **Name and save** card that I already opened

Reply:
Agreed that offering "Name and save" for an already-saved card is inappropriate.  I've added support for DESFire cards to `NfcSceneDeviceInfo` and now go there instead of `NfcSceneReadMifareDesfireSuccess` when reading a saved card.  The "Data" button goes to `NfcSceneMifareDesfireData`, and `NfcSceneMifareDesfireMenu` is now only accessible when scanning a new card.

bf643267394ed3ee8988eabc2af02c64d5b6aee4

![device_info](https://user-images.githubusercontent.com/184975/159061589-ebb4c878-1674-4da0-a6bb-c67fb9cf80a9.png)

I also tried another approach, keeping `NfcSceneReadMifareDesfireSuccess` in both cases, and showing the 'More' button selectively based on scene history.  It's in my `desfire_wip` branch (https://github.com/kevinwallace/flipperzero-firmware/commit/26ef6e03d5c82c66c64935c6e0665f923eb252b0, https://github.com/kevinwallace/flipperzero-firmware/commit/a42c938b6e27c3c952836fbeb20f93e10dab9fc8) if you'd like to take a look, but after trying both, I prefer the approach in this PR.

Reply:
I like current approach with adding DESFire to NfcSceneDeviceInfo scene